lib/study.py

```diff
@@ -0,0 +1,248 @@
+        self.paths["sixtrack_out"] = os.path.join(self.location, "sixtrack_output")
+        app_path = os.path.abspath(inspect.getfile(Study))
+        app_path = os.path.dirname(os.path.dirname(app_path))
+        tem_path = os.path.join(app_path, 'templates')
+        self.paths["templates"] = tem_path
+
+        os.mkdir(self.paths["madx_in"])
+        os.mkdir(self.paths["madx_out"])
+        os.mkdir(self.paths["sixtrack_in"])
+        os.mkdir(self.paths["sixtrack_out"])
+
+        if os.path.isdir(tem_path):
+             for item in os.listdir(tem_path):
+                 s = os.path.join(tem_path, item)
+                 d = os.path.join(self.location, item)
+                 if os.path.isfile(s):
+                     shutil.copy2(s, d)
+        else:
+            print("Invalid templates folder!")
+        self.madx_output = {
+                'fc.2': 'fort.2',
+                'fc.3': 'fort.3.mad',
+                'fc.3.aux': 'fort.3.aux',
+                'fc.8': 'fort.8',
+                'fc.16': 'fort.16',
+                'fc.34': 'fort.34'}
+        self.oneturn_sixtrack_params = {
+                "turnss": 1,
+                "nss": 1,
+                "ax0s": 0.1,
+                "ax1s": 0.1,
+                "imc": 1,
+                "iclo6": 2,
+                "writebins": 1,
+                "ratios": 1,
+                "Runnam": 'FirstTurn',
+                "idfor": 0,
+                "ibtype": 0,
+                "ition": 0,
+                "CHRO": '/',
+                "TUNE": '/',
+                "POST": 'POST',
+                "POS1": '',
+                "ndafi": 1,
+                "tunex": 62.28,
+                "tuney": 60.31,
+                "inttunex": 62.28,
+                "inttuney": 60.31,
+                "DIFF": '/DIFF',
+                "DIF1": '/',
+                "pmass": 938.272013,
+                "emit_beam": 3.75,
+                "e0": 7000,
+                "bunch_charge": 1.15E11,
+                "CHROM": 0,
+                "chrom_eps": 0.000001,
+                "dp1": 0.000001,
+                "dp2": 0.000001,
+                "chromx": 2,
+                "chromy": 2,
+                "TUNEVAL": '/',
+                "CHROVAL": '/'}
+        self.oneturn_sixtrack_input['input'] = self.madx_output
+        self.oneturn_sixtrack_output = ['fort.10']
+        self.sixtrack_output = ['fort.10']
+
+        self.tables = {
+                "mad6t_run": ['seed', 'mad_in', 'mad_out', 'fort.2', 'fort.3',\
+                        'fort.8', 'fort.16', 'job_stdout', 'job_stderr',\
+                        'job_stdlog', 'mad_out_mtime'],
+                "six_input": list(self.sixtrack_params.keys()) + ['id_mad6t_run'],
+                "six_beta": ['seed', 'tunex', 'tuney', 'beta11', 'beta12',\
+                        'beta22', 'beta21', 'qx', 'qy', 'id_mad6t_run'],
+                "dymap_results": []}
+
```

Comment:
I would add a method which checks that key fields are filled, and also another one which echoes all the class field - but these can come later, not necessarily in this pull request